src/TraceParserGen.Tests/TestBase.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.IO;
+using Xunit.Abstractions;
+
+namespace TraceParserGen.Tests
+{
+    /// <summary>
+    /// Base class for TraceParserGen tests
+    /// </summary>
+    public abstract class TestBase
+    {
+        protected static string OriginalInputDir = FindInputDir();
+        protected static string TestDataDir = Path.GetFullPath("inputs");
+        protected static string BaseOutputDir = Path.GetFullPath("output");
+
+        protected TestBase(ITestOutputHelper output)
+        {
+            Output = output;
+            OutputDir = Path.Combine(BaseOutputDir, Guid.NewGuid().ToString("N").Substring(0, 8));
+            
+            // Ensure output directory exists
+            if (!Directory.Exists(OutputDir))
+            {
+                Directory.CreateDirectory(OutputDir);
+            }
+        }
+
+        protected ITestOutputHelper Output { get; }
+
+        protected string OutputDir { get; }
+
+        /// <summary>
+        /// Finds the input directory for test files
+        /// </summary>
+        private static string FindInputDir()
+        {
+            string dir = Environment.CurrentDirectory;
+            while (dir != null)
+            {
+                string candidate = Path.Combine(dir, @"TraceParserGen.Tests\inputs");
+                if (Directory.Exists(candidate))
+                {
+                    return Path.GetFullPath(candidate);
+                }
+
+                dir = Path.GetDirectoryName(dir);
+            }
+            return @"%PERFVIEW%\src\TraceParserGen.Tests\inputs";
+        }
+
+        /// <summary>
+        /// Gets the path to the TraceParserGen.exe executable
+        /// </summary>
+        protected string GetTraceParserGenExePath()
+        {
+            // Search for TraceParserGen.exe in common build output locations
+            string[] searchPaths = new[]
+            {
+                // Relative to test output directory
+                Path.Combine(Environment.CurrentDirectory, @"../../../../TraceParserGen/bin/Release/net462/TraceParserGen.exe"),
```

Comment:
This is not a deterministic way to get TraceParserGen.exe.  Instead, please update the TraceParserGen.Tests.csproj to depend on the TraceParserGen build.  Copy the binary that matches the flavor (debug/release) into the binaries directory when building TraceParserGen.Tests.  Then, you can directly point at the current bin directory for the exe to target rather than searching for it.